vm/src/compile.rs

```diff
@@ -13,14 +13,17 @@ struct Compiler {
     code_object_stack: Vec<CodeObject>,
     nxt_label: usize,
     current_source_location: ast::Location,
+    source_path: Option<String>,
```

Comment:
No need to make source_path a member of compiler, if you attach the sourcepath to the code_object.